src/libstore/filetransfer.cc

```diff
@@ -308,6 +308,9 @@ struct curlFileTransfer : public FileTransfer
 
             curl_easy_setopt(req, CURLOPT_HTTPHEADER, requestHeaders);
 
+            if (settings.downloadSpeed.get() > 0)
+                curl_easy_setopt(req, CURLOPT_MAX_RECV_SPEED_LARGE, (curl_off_t) (settings.downloadSpeed.get() * 1024));
```

Comment:
If this is in bytes/second then a future improvement could be support for prefixes.

Reply:
indeed, this could be useful at some point. But I expect people needing this feature to have a low value less than ~10 MB/s so it still bearable. But this could be a further improvement after this PR.